src/main/java/de/tum/in/www1/artemis/web/rest/ParticipationResource.java

```diff
@@ -352,39 +358,70 @@ public ResponseEntity<ProgrammingExerciseStudentParticipation> resumeParticipati
     @PutMapping("exercises/{exerciseId}/request-feedback")
     @EnforceAtLeastStudent
     @FeatureToggle(Feature.ProgrammingExercises)
-    public ResponseEntity<ProgrammingExerciseStudentParticipation> requestFeedback(@PathVariable Long exerciseId, Principal principal) {
+    public ResponseEntity<StudentParticipation> requestFeedback(@PathVariable Long exerciseId, Principal principal) {
         log.debug("REST request for feedback request: {}", exerciseId);
-        var programmingExercise = programmingExerciseRepository.findByIdWithTemplateAndSolutionParticipationElseThrow(exerciseId);
 
-        if (programmingExercise.isExamExercise()) {
-            throw new BadRequestAlertException("Not intended for the use in exams", "participation", "preconditions not met");
+        Exercise exercise = exerciseRepository.findById(exerciseId).orElseThrow(() -> new ResourceNotFoundException("Exercise not found"));
+
+        if (!(exercise instanceof TextExercise) && !(exercise instanceof ProgrammingExercise)) {
+            throw new BadRequestAlertException("Unsupported exercise type", "participation", "unsupported type");
         }
 
-        if (programmingExercise.getDueDate() != null && now().isAfter(programmingExercise.getDueDate())) {
+        return handleExerciseFeedbackRequest(exercise, principal);
+    }
+
+    private ResponseEntity<StudentParticipation> handleExerciseFeedbackRequest(Exercise exercise, Principal principal) {
```

Comment:
Please move this method or parts of it into the corresponding service

Reply:
Everything related to ParticipationResource will be improved on a follow up Pr which will also add athena feedback for modeling exercises.